kodev

```diff
@@ -981,6 +981,69 @@ OPTIONS:
     } && popd || exit
 }
 
+function kodev-check() {
+    exit_code=0
+    check_submodules
+
+    # shellcheck disable=2016
+    mapfile -t shellscript_locations < <({ git grep -lE '^#!(/usr)?/bin/(env )?(bash|sh)' && git submodule --quiet foreach '[ "$path" = "base" -o "$path" = "platform/android/luajit-launcher" ] || git grep -lE "^#!(/usr)?/bin/(env )?(bash|sh)" | sed "s|^|$path/|"' && git ls-files ./*.sh; } | sort | uniq)
```

Comment:
Does anyone know what exactly the `git grep -lE '^#!(/usr)?/bin/(env )?(bash|sh)'` is supposed to pick up?

When run from the `.ci/` folder i get all the scripts inside it (that are also picked up by the other ones). This is the way CircleCi uses it.
If i run it locally directly in the repo I get tons of other files including many from crengine. however GHA doe not seem to pick u on those (it should also be running from the koreader folder).

Which set of files do we actually want with this?

Reply:
git blame should probably say why I wrote it.

> If i run it locally directly in the repo I get tons of other files including many from crengine.

I can't confirm that? Could you post the full command and output? Afaik it's been the same from git 2.1x or so all the way up to 2.32.

```
koreader$ ({ git grep -lE '^#!(/usr)?/bin/(env )?(bash|sh)' && git submodule --quiet foreach '[ "$path" = "base" -o "$path" = "platform/android/luajit-launcher" ] || git grep -lE "^#!(/usr)?/bin/(env )?(bash|sh)" | sed "s|^|$path/|"' && git ls-files ./*.sh; } | sort | uniq)
.ci/after_success.sh
.ci/before_install.sh
.ci/build.sh
.ci/check.sh
.ci/common.sh
.ci/fetch.sh
.ci/helper_luarocks.sh
.ci/helper_shellchecks.sh
.ci/install.sh
.ci/script.sh
.ci/test.sh
kodev
platform/appimage/AppRun
platform/cervantes/disable-usbms.sh
platform/cervantes/disable-wifi.sh
platform/cervantes/enable-usbms.sh
platform/cervantes/enable-wifi.sh
platform/cervantes/koreader.sh
platform/cervantes/koreader-standalone.sh
platform/cervantes/obtain-ip.sh
platform/cervantes/release-ip.sh
platform/cervantes/restore-wifi-async.sh
platform/cervantes/resume.sh
platform/cervantes/spinning_zsync
platform/cervantes/suspend.sh
platform/common/spinning_zsync
platform/debian/do_debian_package.sh
platform/debian/koreader.sh
platform/kindle/extensions/koreader/bin/koreader-ext.sh
platform/kindle/koreader.sh
platform/kindle/libkohelper.sh
platform/kobo/disable-wifi.sh
platform/kobo/enable-wifi.sh
platform/kobo/fmon/koreader.sh
platform/kobo/koreader.sh
platform/kobo/nickel.sh
platform/kobo/obtain-ip.sh
platform/kobo/release-ip.sh
platform/kobo/restore-wifi-async.sh
platform/mac/do_mac_bundle.sh
platform/pocketbook/koreader.app
platform/pocketbook/system_koreader.app
platform/remarkable/disable-wifi.sh
platform/remarkable/enable-wifi.sh
platform/remarkable/koreader.sh
platform/sony-prstux/resume.sh
platform/sony-prstux/set-wifi.sh
platform/sony-prstux/suspend.sh
platform/ubuntu-touch/koreader.sh
plugins/backgroundrunner.koplugin/luawrapper.sh
plugins/backgroundrunner.koplugin/wrapper.sh
```

It was intended to be as inclusive as possible. See https://github.com/koreader/koreader/pull/5096#issuecomment-506953268 as for why luajit-launcher is excluded a bit more draconically than I intended.

In retrospect I should've probably suggested reducing it to `git grep -lE '^#!(/usr)?/bin/(env )?(bash|sh)'`, but I really only meant to exclude the gradle script, not our own. Much the same applies for base really. At the time that wasn't ready to pass yet.

Of course in principle that should all be checked in the relevant submodules themselves, but I liked the idea of a final check on everything.

tl;dr You can probably safely reduce the line to `git grep -lE '^#!(/usr)?/bin/(env )?(bash|sh)'`.

Reply:
koreader]$ { git grep -lE '^#!(/usr)?/bin/(env )?(bash|sh)' && git submodule --quiet foreach '[ "$path" = "base" -o "$path" = "platform/android/luajit-launcher" ] || git grep -lE "^#!(/usr)?/bin/(env )?(bash|sh)" | sed "s|^|$path/|"' && git ls-files ./*.sh; } | sort | uniq
base/.ci/build_script.sh
base/.ci/common.sh
base/.ci/emulator_install.sh
base/.ci/helper_luarocks.sh
base/.ci/helper_shellchecks.sh
base/.ci/lint_install.sh
base/.ci/lint_script.sh
base/.ci/test_script.sh
base/thirdparty/cmake_modules/patch-wrapper.sh
base/thirdparty/kpvcrlib/crengine/.ci/script.sh
base/thirdparty/kpvcrlib/crengine/cr3gui/po/update-po
base/thirdparty/kpvcrlib/crengine/cr3wx/macros/autogen.sh
base/thirdparty/kpvcrlib/crengine/thirdparty/antiword/Docs/Mozilla
base/thirdparty/kpvcrlib/crengine/thirdparty/antiword/kantiword
base/thirdparty/kpvcrlib/crengine/thirdparty/antiword/Unix-only/KDE1-only/kantiword.sh
base/thirdparty/kpvcrlib/crengine/thirdparty/antiword/Unix-only/KDE3-only/kantiword.sh
base/thirdparty/kpvcrlib/crengine/thirdparty/chmlib/contrib/mozilla_helper.sh
base/thirdparty/kpvcrlib/crengine/thirdparty/chmlib/install-sh
base/thirdparty/kpvcrlib/crengine/thirdparty/freetype/autogen.sh
base/thirdparty/kpvcrlib/crengine/thirdparty/freetype/builds/atari/gen-purec-patch.sh
base/thirdparty/kpvcrlib/crengine/thirdparty/freetype/builds/unix/install-sh
base/thirdparty/kpvcrlib/crengine/thirdparty/freetype/configure
base/thirdparty/kpvcrlib/crengine/thirdparty/libpng/install-sh
base/thirdparty/kpvcrlib/crengine/thirdparty/zlib/configure
base/utils/pdfattach
.ci/after_success.sh
.ci/before_install.sh
.ci/build.sh
.ci/check.sh
.ci/common.sh
.ci/fetch.sh
.ci/helper_luarocks.sh
.ci/helper_shellchecks.sh
.ci/install.sh
.ci/script.sh
.ci/test.sh
kodev
platform/android/luajit-launcher/gradlew
platform/android/luajit-launcher/jni/luajit/mk-luajit.sh
platform/appimage/AppRun
platform/cervantes/disable-usbms.sh
platform/cervantes/disable-wifi.sh
platform/cervantes/enable-usbms.sh
platform/cervantes/enable-wifi.sh
platform/cervantes/koreader.sh
platform/cervantes/koreader-standalone.sh
platform/cervantes/obtain-ip.sh
platform/cervantes/release-ip.sh
platform/cervantes/restore-wifi-async.sh
platform/cervantes/resume.sh
platform/cervantes/spinning_zsync
platform/cervantes/suspend.sh
platform/common/spinning_zsync
platform/debian/do_debian_package.sh
platform/debian/koreader.sh
platform/kindle/extensions/koreader/bin/koreader-ext.sh
platform/kindle/koreader.sh
platform/kindle/libkohelper.sh
platform/kobo/disable-wifi.sh
platform/kobo/enable-wifi.sh
platform/kobo/fmon/koreader.sh
platform/kobo/koreader.sh
platform/kobo/nickel.sh
platform/kobo/obtain-ip.sh
platform/kobo/release-ip.sh
platform/kobo/restore-wifi-async.sh
platform/mac/do_mac_bundle.sh
platform/pocketbook/koreader.app
platform/pocketbook/system_koreader.app
platform/remarkable/disable-wifi.sh
platform/remarkable/enable-wifi.sh
platform/remarkable/koreader.sh
platform/sony-prstux/resume.sh
platform/sony-prstux/set-wifi.sh
platform/sony-prstux/suspend.sh
platform/ubuntu-touch/koreader.sh
plugins/backgroundrunner.koplugin/luawrapper.sh
plugins/backgroundrunner.koplugin/wrapper.sh

Reply:
```
git --version 
git version 2.34.1
```

Reply:
Which shell? It's clearly not excluding base and luajit-launcher like it should.